libvast/src/system/index.cpp

```diff
@@ -600,6 +600,9 @@ void index_state::schedule_lookups() {
       auto it = pending_queries.queries().find(qid);
       if (it == pending_queries.queries().end()) {
         VAST_WARN("{} tried to access non-existant query {}", *self, qid);
+        *cnt -= 1;
+        if (*cnt == 0)
+          --running_partition_lookups;
```

Comment:
This snippet is now duplicated between here and `handle_completion` directly below. Can we put it into a lambda to give it a name?

E.g.,

```cpp
auto finish_query = [&] {
  if (--*cnt > 0)
    return false;
  --running_partition_lookups;
  return true;
};
```

Reply:
You need to be careful with that. We call `handle_completion` when we get the response from the partition. If we create this helper closure outside of it the reference to `cnt` won't be valid later on. You would have to capture `cnt` by value and `finish_query` also by value in `handle_completion`.